### ISSUE TITLE
[Network Issue] Got 504 response and chat keep "thinking" for a long time

### ISSUE THREAD

**Issue by LikeDreamwalker** [microsoft/vscode-copilot-release#1027]
- VS Code Version: latest
- OS Version: Windows 11
- Logs:
`2024-03-13 00:53:58.503 [info] [auth] Got Copilot token for LikeDreamwalker
2024-03-13 00:54:07.566 [info] [chat fetch] url https://api.githubcopilot.com/chat/completions
2024-03-13 00:54:07.566 [info] [chat fetch] modelMaxPromptTokens 3072
2024-03-13 00:54:07.566 [info] [chat fetch] modelMaxResponseTokens 2201
2024-03-13 00:54:07.566 [info] [chat fetch] chat model gpt-4
2024-03-13 00:56:09.896 [info] [chat fetch] request.response: [https://api.githubcopilot.com/chat/completions], took 122328 ms
2024-03-13 00:56:09.900 [info] [chat fetch] Request ID for failed request: ,d51aafaa-3ef0-4ed1-b259-df8df1e02a53
2024-03-13 01:00:58.258 [info] [chat fetch] url https://api.githubcopilot.com/chat/completions
2024-03-13 01:00:58.258 [info] [chat fetch] modelMaxPromptTokens 3072
2024-03-13 01:00:58.258 [info] [chat fetch] modelMaxResponseTokens 2201
2024-03-13 01:00:58.258 [info] [chat fetch] chat model gpt-4
2024-03-13 01:03:01.777 [info] [chat fetch] request.response: [https://api.githubcopilot.com/chat/completions], took 123519 ms
2024-03-13 01:03:01.779 [info] [chat fetch] Request ID for failed request: ,aea37b9c-2873-44b8-ad5c-7689767d1243`

![image](https://github.com/microsoft/vscode-copilot-release/assets/45194556/98f5f5b9-0a27-4358-aa21-15ce7aecdd73)

Also, I tried to ping api.githubcopilot.com, but the response is pretty bad, and even if I use a proxy seems like it won't help
![image](https://github.com/microsoft/vscode-copilot-release/assets/45194556/3242cf79-eded-4c38-a696-ac4b21c45974)

I didn't encounter this issue until months ago. I tried many ways, like presetting a specific proxy in vscode, and sometimes it worked. Now, this issue can be met very often, and even I am using codespaces, which I guess is not connected with my client directly but should be the codespaces server.

**Comment by lramos15** [comment]
Are you still experiencing this? Does it happen often? Wondering if this is a service incident or something else

**Comment by LikeDreamwalker** [comment]
> Are you still experiencing this? Does it happen often? Wondering if this is a service incident or something else

I tried two of my products, and now it seems like everything is fine, but I think this issue might happen again.
I can't tell if this issue is based on a server issue cause it seems like the connection between client and server goes wrong, and I don't know why. This might help that I meet this issue more often when I am using local vscode rather than codespaces. And not all the time I can get a 504 back, mostly a long time "thinking" which I cannot wait anymore.
I also considered this might be a network issue, but when this issue happened, I was able to use GitHub, until upper I tried ping the host directly, and it really responded badly. And now the server responds steadily enough.
![image](https://github.com/microsoft/vscode-copilot-release/assets/45194556/4b973401-42b8-46c0-888d-378d76fd0dee)

And I can reply with requestID on this issue when I meet this issue again, maybe that can help you guys figure out what's wrong.

Thanks for your reply.
